dev/ProgressRing/ProgressRing.cpp

```diff
@@ -41,18 +47,98 @@ void ProgressRing::OnRangeBasePropertyChanged(const winrt::DependencyObject&, co
     UpdateSegment();
 }
 
+void ProgressRing::OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
+void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
 void ProgressRing::OnStrokeThicknessPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
+    ApplyLottieAnimation();
     UpdateRing();
 }
 
-winrt::Size ProgressRing::ComputeEllipseSize(double thickness, double actualWidth, double actualHeight)
+void ProgressRing::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressRing::ApplyLottieAnimation()
+{
+    if (auto&& player = m_player.get())
+    {
+        // ProgressRing only accounts for ActualWidth to ensure that it is always a circle
```

Comment:
Comments should end in periods. What does this comment mean though... that it doesn't account for ActualHeight, or that it use using ActualWidth instead of Width, or something else? Seems like a message for yourself rather than the next reader.

Reply:
It only accounts for width, rather than width + height so that it is a circle rather than an ellipse. Let me reword the comment.